auth/grpctransport/directpath.go

```diff
@@ -121,3 +128,37 @@ func configureDirectPath(grpcOpts []grpc.DialOption, opts *Options, endpoint str
 	}
 	return grpcOpts, endpoint
 }
+
+// OnComputeEngine returns whether the client is running on GCE.
+//
+// It is a copy of the gRPC internal OnGCE() func at:
+// https://github.com/grpc/grpc-go/blob/master/internal/googlecloud/googlecloud.go
+func OnComputeEngine() bool {
```

Comment:
Why is this a public func? I think it's worth considering whether it and its supporting code should be located in a new package inside the top-level `internal` directory and made more broadly available for re-use with this library, but hidden from public usage.

Reply:
SG, I moved this to internal/compute/

plz let me know if you prefer other package name, thanks:)